hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -744,11 +744,20 @@ public void runMetaSync() {
             props.getInteger(HoodieIndexConfig.BUCKET_INDEX_NUM_BUCKETS.key())));
       }
 
+      //Collect exceptions in list because we want all sync to run. Then we can throw
+      List<HoodieException> metaSyncExceptions = new ArrayList<>();
       for (String impl : syncClientToolClasses) {
-        Timer.Context syncContext = metrics.getMetaSyncTimerContext();
-        SyncUtilHelpers.runHoodieMetaSync(impl.trim(), metaProps, conf, fs, cfg.targetBasePath, cfg.baseFileFormat);
-        long metaSyncTimeMs = syncContext != null ? syncContext.stop() : 0;
-        metrics.updateDeltaStreamerMetaSyncMetrics(getSyncClassShortName(impl), metaSyncTimeMs);
+        try {
+          Timer.Context syncContext = metrics.getMetaSyncTimerContext();
+          SyncUtilHelpers.runHoodieMetaSync(impl.trim(), metaProps, conf, fs, cfg.targetBasePath, cfg.baseFileFormat);
+          long metaSyncTimeMs = syncContext != null ? syncContext.stop() : 0;
+          metrics.updateDeltaStreamerMetaSyncMetrics(getSyncClassShortName(impl), metaSyncTimeMs);
+        } catch (HoodieException e) {
+          metaSyncExceptions.add(e);
```

Comment:
Let's log the error and full stack trace here itself instead of doing printStackTrace.